Fix #1630: Bookmark Validation

Brandon-T opened the pull request:
## Summary of Changes

This pull request fixes issue https://github.com/brave/brave-ios/issues/1630

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
<!-- Any useful notes explaining how best to test and verify. -->


## Screenshots:
<!-- If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here. -->


## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

Brandon-T committed:
Fixes bookmark validation.

iccub reviewed (changes requested):
fix tests, se CI logs for more details

Brandon-T committed:
Stricter unit tests and validation to NOT mess up validation with bookmarks vs. bookmarklets.
Bookmarks are a sensitive topic and these unit tests will help determine what is valid and what isn't.

Brandon-T commented:
@iccub I added more unit tests and stricter validation. This helped me catch quite a few bugs. Should be much better now.

iccub commented:
tests still failing

Brandon-T committed:
Fixed last test error.
